From fe09fefa60dd59843e99b714e5e00924a5829db6 Mon Sep 17 00:00:00 2001
From: Lu Hui <luhui@sipeed.com>
Date: Mon, 6 Nov 2023 16:35:47 +0800
Subject: [PATCH 17/53] drivers: cw2015: be quite, dont repeat warn

---
 drivers/power/supply/cw2015_battery.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/power/supply/cw2015_battery.c b/drivers/power/supply/cw2015_battery.c
index de1fa71be..47e4cde1b 100644
--- a/drivers/power/supply/cw2015_battery.c
+++ b/drivers/power/supply/cw2015_battery.c
@@ -347,7 +347,8 @@ static void cw_update_charge_status(struct cw_battery *cw_bat)
 
 	ret = power_supply_am_i_supplied(cw_bat->rk_bat);
 	if (ret < 0) {
-		dev_warn(cw_bat->dev, "Failed to get supply state: %d\n", ret);
+		// I known, dont repeat
+		// dev_warn(cw_bat->dev, "Failed to get supply state: %d\n", ret);
 	} else {
 		bool charger_attached;
 
-- 
2.34.1

