From bba540473807edd3dde313f9a0604dfd673e5eef Mon Sep 17 00:00:00 2001
From: Lu Hui <luhui@sipeed.com>
Date: Mon, 13 Nov 2023 11:52:10 +0800
Subject: [PATCH 25/53] drivers: add 1920x1080_60Hz dsi panel

---
 drivers/gpu/drm/panel/panel-simple.c | 32 ++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index f9986a549..f6a92cd9b 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -4741,6 +4741,35 @@ static const struct panel_desc_dsi z14inch_1920x1200_54Hz = {
 	.lanes = 4,
 };
 
+static const struct drm_display_mode z15inch_1920x1080_60Hz_mode = {
+	.clock = 138500,
+	.hdisplay = 1920,
+	.hsync_start = 1920 + 48,
+	.hsync_end = 1920 + 48 + 32,
+	.htotal = 1920 + 48 + 32 + 80,
+	.vdisplay = 1080,
+	.vsync_start = 1080 + 3,
+	.vsync_end = 1080 + 3 + 5,
+	.vtotal = 1080 + 3 + 5 + 23,
+	.flags = DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC,
+};
+
+static const struct panel_desc_dsi z15inch_1920x1080_60Hz = {
+	.desc = {
+		.modes = &z15inch_1920x1080_60Hz_mode,
+		.num_modes = 1,
+		.bpc = 8,
+		.size = {
+			.width = 62,
+			.height = 110,
+		},
+		.connector_type = DRM_MODE_CONNECTOR_DSI,
+	},
+	.flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_VIDEO_BURST,
+	.format = MIPI_DSI_FMT_RGB888,
+	.lanes = 4,
+};
+
 static const struct of_device_id dsi_of_match[] = {
 	{
 		.compatible = "auo,b080uan01",
@@ -4769,6 +4798,9 @@ static const struct of_device_id dsi_of_match[] = {
 	}, {
 		.compatible = "custom,z14inch_1920x1200_54Hz",
 		.data = &z14inch_1920x1200_54Hz
+	}, {
+		.compatible = "custom,z15inch_1920x1080_60Hz",
+		.data = &z15inch_1920x1080_60Hz
 	}, {
 		/* sentinel */
 	}
-- 
2.34.1

