From 3cc3d95034a8ab593d2a4e9d4cddb6a4cef36e0c Mon Sep 17 00:00:00 2001
From: Lu Hui <luhui@sipeed.com>
Date: Thu, 16 Nov 2023 10:18:31 +0800
Subject: [PATCH 29/53] drivers: panel: add 2160x1440_40Hz dsi panel

---
 drivers/gpu/drm/panel/panel-simple.c | 32 ++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index f6a92cd9b..2d841050b 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -4712,6 +4712,35 @@ static const struct panel_desc_dsi hlt_hpk070h275 = {
 	.lanes	= 4,
 };
 
+static const struct drm_display_mode z14inch_2160x1440_40Hz_mode = {
+	.clock          = 136000,
+	.hdisplay       = 2160,
+	.hsync_start    = 2160 + 48,
+	.hsync_end      = 2160 + 48 + 32,
+	.htotal         = 2160 + 48 + 32 + 80,
+	.vdisplay       = 1440,
+	.vsync_start    = 1440 + 3,
+	.vsync_end      = 1440 + 3 + 10,
+	.vtotal         = 1440 + 3 + 10 + 14,
+        .flags = DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC,
+};
+
+static const struct panel_desc_dsi z14inch_2160x1440_40Hz = {
+        .desc = {
+                .modes = &z14inch_2160x1440_40Hz_mode,
+                .num_modes = 1,
+                .bpc = 8,
+                .size = {
+                        .width = 62,
+                        .height = 110,
+                },
+                .connector_type = DRM_MODE_CONNECTOR_DSI,
+        },
+        .flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_VIDEO_BURST,
+        .format = MIPI_DSI_FMT_RGB888,
+        .lanes = 4,
+};
+
 static const struct drm_display_mode z14inch_1920x1200_54Hz_mode = {
         .clock = 138250,
         .hdisplay = 1920,
@@ -4795,6 +4824,9 @@ static const struct of_device_id dsi_of_match[] = {
 	}, {
 		.compatible = "hlt,hpk070h275",
 		.data = &hlt_hpk070h275
+	}, {
+		.compatible = "custom,z14inch_2160x1440_40Hz",
+		.data = &z14inch_2160x1440_40Hz
 	}, {
 		.compatible = "custom,z14inch_1920x1200_54Hz",
 		.data = &z14inch_1920x1200_54Hz
-- 
2.34.1

