From a819e3e119126892675ee5c7c720317b05005a31 Mon Sep 17 00:00:00 2001
From: Lu Hui <luhui@sipeed.com>
Date: Fri, 24 Nov 2023 18:52:52 +0800
Subject: [PATCH 45/53] sound: es8156: limit max volume to 170

---
 sound/soc/codecs/es8156.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/es8156.c b/sound/soc/codecs/es8156.c
index 47ab55ef0..4986d4889 100644
--- a/sound/soc/codecs/es8156.c
+++ b/sound/soc/codecs/es8156.c
@@ -114,8 +114,9 @@ static const struct snd_kcontrol_new es8156_snd_controls[] = {
 	SOC_DOUBLE("ALC Maximum Minimum Volume",ES8156_ALC_CONFIG3_REG17,
 	4,0,15,0),
 	/* DAC Digital controls */
+// TODO: we need change this limit from devicetree
 	SOC_SINGLE_TLV("DAC Playback Volume", ES8156_VOLUME_CONTROL_REG14,
-			  0, 0xff, 0, dac_vol_tlv),
+			  0, 170, 0, dac_vol_tlv),
 	SOC_SINGLE("HP Switch",ES8156_ANALOG_SYS3_REG22,3,1,0),
 
 
@@ -188,7 +189,8 @@ static const struct snd_soc_dapm_route es8156_dapm_routes[] = {
 #define MCLK_SOURCE			SCLK_PIN		// select MCLK source, MCLK_PIN or SCLK_PIN
 #define EQ7bandOn			0
 #define VDDA_VOLTAGE		VDDA_3V3
-#define DAC_Volume			191				// DAC digital gain
+// TODO: we need change this value from devicetree
+#define DAC_Volume	170	// DAC digital gain
 #define DACHPModeOn			0				// disable:0, enable:1
 
 /**************************************************/
-- 
2.34.1

